python-stdlib/tarfile-write/tarfile/write.py

```diff
@@ -67,7 +67,7 @@ def addfile(self, tarinfo, fileobj=None):
             name += "/"
     hdr = uctypes.struct(uctypes.addressof(buf), _TAR_HEADER, uctypes.LITTLE_ENDIAN)
     hdr.name[:] = name.encode("utf-8")[:100]
-    hdr.mode[:] = b"%07o\0" % (tarinfo.mode & 0o7777)
+    hdr.mode[:] = b"%07o\0" % (0o0755 & 0o7777)
```

Comment:
I think this change can be dropped, yeah? Otherwise all tar contents will have RWX permissions.

Reply:
unfortunately `tarinfo.mode` takes the `0x4000` or `0x8000` depending on wether the resource is a file or a folder, turning the file permission matrix into `0o4000` or `0o8000`, hence no permissions from anyone.
The MicroPython file system does not return permissions in `stat[0]`, as it happens for python on a PC.
This would lead to archives that require multiple permissions changes from the user, especially when compressing folder structures

Reply:
Oh, right, gotcha. I should have looked up the numeric values of `_S_IFDIR` and `_S_IF_REG` .

What about `(0o755 if tarinfo.mode == S_IFDIR else 0o644)` for this argument, will that give reasonable results without setting execute on normal files?

Reply:
@projectgus I initially tried setting `0o644` for everything and it didn't work, but will try this one and update the PR :)

Thank you for taking the time to follow up

Reply:
@projectgus , now I have a problem: `S_IFDIR` is not recognised within `write.py`.
I guess it's because it's a private member of `tarfile` defined in `__init__.py`

I'm not so deeply skilled in Python to know how to get around that, learning as I go :D 

```
>>> tw.add('test_folder')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/lib/tarfile/write.py", line 115, in add
  File "/lib/tarfile/write.py", line 70, in addfile
NameError: name 'S_IFDIR' isn't defined
```

I found out that this works, though
`(0o755 if tarinfo.isdir() else 0o644)`

would it be an acceptable solution?

Reply:
@ubidefeo That version looks even better to me actually, good idea!